sleap_nn/data/confidence_maps.py

```diff
@@ -0,0 +1,74 @@
+"""Generate confidence maps."""
+from torch.utils.data.datapipes.datapipe import IterDataPipe
+from typing import Optional
+import sleap_io as sio
+import torch
+
+
+def make_confmaps(
+    points: torch.Tensor, xv: torch.Tensor, yv: torch.Tensor, sigma: float
+):
+    """Create confidence maps from points and grid vectors."""
+    x = torch.reshape(points[:, 0], (1, 1, -1))
+    y = torch.reshape(points[:, 1], (1, 1, -1))
+    cm = torch.exp(
+        -(
+            (torch.reshape(xv, (1, -1, 1)) - x) ** 2
+            + (torch.reshape(yv, (-1, 1, 1)) - y) ** 2
+        )
+        / (2 * sigma**2)
+    )
+
+    # Replace NaNs with 0.
+    cm = torch.where(torch.isnan(cm), 0.0, cm)
+    return cm
+
+
+def make_grid_vectors(image_height: int, image_width: int, output_stride: int):
+    """Create grid vectors of appropriate shape and stride."""
+    xv = torch.arange(0, image_width, step=output_stride).to(
+        torch.float32
+    )  # (image_width,)
+    yv = torch.arange(0, image_height, step=output_stride).to(
+        torch.float32
+    )  # (image_height,)
+    return xv, yv
+
+
+class ConfidenceMapGenerator(IterDataPipe):
+    """DataPipe for generating confidence maps.
+
+    This DataPipe will generate confidence maps for examples from the input pipeline.
+
+    Attributes:
+        source_dp: The input `IterDataPipe` with examples that contain an instance and
+            an image.
+        sigma: The standard deviation of the Gaussian distribution that is used to
+            generate confidence maps.
+        output_stride: The relative stride to use when generating confidence maps.
```

Comment:
Document what keys must be present in the input examples and what shapes they're expected to have.

Also maybe make `instance_key: str = "instance_image"` another keyword argument so this can be modified for different pipelines

Reply:
Also `image_key: str = "instance_image"`